src/frontend/AppContainerWrapper.tsx

```diff
@@ -92,13 +99,42 @@ function hidePracticeModeTemporarily(route: string | null) {
   return route !== null && TEMP_HIDE_PRACTICE_MODE_UI.includes(route);
 }
 
+const AuthContainer = createAppContainer(AuthStack);
+
 const AppContainerWrapper = () => {
   const navRef = React.useRef<NavigationContainerComponent>();
   const [inviteModalEnabled, setInviteModalEnabled] = React.useState(true);
   const [queuedInvite, setQueuedInvite] = React.useState<string | null>(null);
   const [{ experiments }] = React.useContext(SettingsContext);
   const [hidePracticeBar, setHidePracticeBar] = React.useState(true);
   const [hidePracticeMode, setHidePracticeMode] = React.useState(false);
+  const [{ authStatus }, setAuthState] = React.useContext(SecurityContext);
+  const isMounted = React.useRef(false);
+
+  React.useEffect(() => {
+    isMounted.current = true;
+  }, []);
+
+  React.useEffect(() => {
+    const appStateListener = AppState.addEventListener(
+      "change",
+      handleStateAndSetCheckPassFlag
+    );
+
+    return () => appStateListener.remove();
+  }, [authStatus, navRef.current]);
```

Comment:
remove `navRef.current` here because adding refs to effects deps is generally never needed/recommended (https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/)